qiskit/visualization/circuit/_utils.py

```diff
@@ -506,32 +506,34 @@ def _sorted_nodes(dag_layer):
 
 
 def _get_gate_span(qubits, node, measure_arrows):
-    """Get the list of qubits drawing this gate would cover
-    qiskit-terra #2802
-    """
-    min_index = len(qubits)
-    max_index = 0
-    for qreg in node.qargs:
-        index = qubits.index(qreg)
-
-        if index < min_index:
-            min_index = index
-        if index > max_index:
-            max_index = index
+    """Return the ordered wires this node would occupy when drawn."""
 
     if isinstance(node.op, ControlFlowOp) and not isinstance(node.op, BoxOp):
         # Because of wrapping boxes for mpl control flow ops, this
         # type of op must be the only op in the layer
         # BoxOps are excepted because they have one block executed unconditionally
-        span = qubits
-    elif node.cargs and (
-        (measure_arrows and isinstance(node.op, Measure)) or getattr(node.op, "condition", None)
+        return qubits
+
+    qubit_index_map = {bit: idx for idx, bit in enumerate(qubits)}
+    qubit_indices = [qubit_index_map[qarg] for qarg in node.qargs if qarg in qubit_index_map]
+
+    if not qubit_indices and not node.cargs:
+        # Operations with no wires shouldn't collide with anything.
+        return []
+
+    if not qubit_indices:
+        # Classical-only operations conservatively reserve the circuit width.
+        return qubits
```

Comment:
I suggest do keep this simplest for the moment. Do we even have operations with classical bits but no quantum bits? We do have the `GlobalPhaseGate` which has neither quantum bits nor classical bits -- and currently does not get drawn, in which case we should indeed return `[]`.

```suggestion
    if not qubit_indices:
        # We should not have operations with only classical wires,
        # and the operations with no qubits shouldn't collide with anything.
        return []
```